src/main/java/com/aws/greengrass/easysetup/DeviceProvisioningHelper.java

```diff
@@ -460,18 +461,28 @@ public void setupIoTRoleForTes(String roleName, String roleAliasName, String cer
             return Optional.of(iamClient.getPolicy(software.amazon.awssdk.services.iam.model.GetPolicyRequest.builder()
                     .policyArn(String.format(MANAGED_IAM_POLICY_ARN_FORMAT, partition, policyName)).build()).policy()
                     .arn());
-        } catch (ResourceNotFoundException | NoSuchEntityException mnf) {
-            // Check if a customer policy exists with the name
-            try {
-                return Optional.of(iamClient.getPolicy(
-                        software.amazon.awssdk.services.iam.model.GetPolicyRequest.builder()
-                                .policyArn(String.format(IAM_POLICY_ARN_FORMAT, partition, getAccountId(), policyName))
-                                .build())
-                        .policy().arn());
-            } catch (ResourceNotFoundException | NoSuchEntityException cnf) {
-                return Optional.empty();
+        } catch (NoSuchEntityException mnf) {
+            outStream.println("No managed IAM policy found, looking for user defined policy...");
+        } catch (IamException e) {
```

Comment:
we get an IamException and not some more specific 403 error?

Reply:
Right, we don't get a specific error, have to look at the error message of the IAM exception